### ISSUE TITLE
On magic numbers

### ISSUE THREAD

**Issue by keke101** [nus-cs2113-AY2021S1/forum#37]
Hello, can I check if the following array indexes used are deemed as magic numbers?
```
String name = arguments[0];
String date = arguments[1];
```
My understanding is that this is pretty straight forward as I'm assigning them directly to two variables which have descriptive names.

**Comment by zhixiangteoh** [comment]
Honestly think this should be perfectly sound and acceptable, although it isn't exactly specified in most places I've looked (including all the [reference links at the bottom of the SE-EDU Java coding standard page](https://se-education.org/guides/conventions/java/basic.html#references)). 

The alternative to indexing your array would be to write something like `int nameIndex = 0; int dateIndex = 1` which for a 2-element array feels like it actually makes the code a little less readable.

A _possibly_ relevant section I found is in [this doc](https://www.oracle.com/technetwork/java/codeconventions-150003.pdf) under section 10.3, but no explicit mention of array indexing.

Agree this is not so clear cut, so would also like to see what the general recommendation is.
